### PR TITLE
[Merged by Bors] - Don't count OSK as an application

### PR DESCRIPTION
**Pull request by AlanGriffiths**
Fixes: #109

### CODE REVIEW

**Comment by AlanGriffiths** [comment]
@Saviq (when you have time) @graysonguarino's tweak works for me, but we need an approver

**Comment by wmww** [comment]
I'd argue that the depth layer is less relevant to if the window should effect the diagnostic, and the real thing you should be looking at is if the window state is attached

**Comment by AlanGriffiths** [comment]
> I'd argue that the depth layer is less relevant to if the window should effect the diagnostic, and the real thing you should be looking at is if the window state is attached

Makes sense (but I'm at EOD)

**Comment by AlanGriffiths** [comment]
Oops! There are scenarios this breaks:

1. start frame & wait for diagnostic
2. start gnome-chess
3. stop gnome-chess

Expect: greyscale (until timeout)
Actual: diagnostic

(Not broken on `main`)

**Comment by AlanGriffiths** [comment]
OK, reworked and working (I think)

File: src/frame_window_manager.cpp [hunk 1]
```diff
@@ -77,6 +77,19 @@ bool override_state(WindowSpecification& spec, WindowInfo const& window_info)
...
+
+auto is_application(WindowInfo const& window_info)
+{
+    switch (window_info.depth_layer())
+    {
+    case mir_depth_layer_application:
+    case mir_depth_layer_always_on_top:
+        return window_info.state() != mir_window_state_attached;
+
+    default:;
```
Comment (graysonguarino) [inline src/frame_window_manager.cpp:89]: Remove semicolon after `default:`

**Review by graysonguarino — APPROVED** [review]
Approving, but with one minor suggestion (unless this is intentional for behavior I don't know about.)

bors delegate

**Comment by AlanGriffiths** [comment]
bors merge

**Review by graysonguarino — APPROVED** [review]
bors merge

**Comment by AlanGriffiths** [comment]
bors retry

**Pull request closed without merging**